receiver/kafkametricsreceiver/consumer_scraper.go

```diff
@@ -143,15 +142,13 @@ func (s *consumerScraper) scrape(context.Context) (pdata.Metrics, error) {
 					break
 				}
 			}
-			labels.UpsertString(metadata.A.Topic, topic)
 			if isConsumed {
 				var lagSum int64
 				var offsetSum int64
 				for partition, block := range partitions {
-					labels.UpsertInt(metadata.A.Partition, int64(partition))
 					consumerOffset := block.Offset
 					offsetSum += consumerOffset
-					addIntGauge(ilm.Metrics(), metadata.M.KafkaConsumerGroupOffset.Name(), now, labels, consumerOffset)
+					s.mb.RecordKafkaConsumerGroupOffsetDataPoint(now, consumerOffset, group.GroupId, topic, fmt.Sprintf("%d", partition))
```

Comment:
Hm this look like a breaking change caused by metrics builder limitation. Even if we potentially can accept this particular int->string change, I believe we should enhance the metrics builder to allow attributes of different types.

Do you think you can add this improvement to the metrics builder before merging this PR?

Reply:
Submitted an issue https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8127

Reply:
Hey @dmitryax I will work on changing metrics builder. 
Do we want to merge this PR as-is and then fix this as part of #8127? 
I am fine either way.

Reply:
I would prefer avoid introducing breaking changes, so i think we should rather hold off on this one until https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8127 is resolved